### ISSUE TITLE
[$1000] [HOLD for C+ Checklist] Dev: Room - Settings - Non members appears in list of room in offline mode

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22353]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. go to Settings > Workspaces
2. click on new workspace button
3. add a few members to workspace
4. go to offline
5. click on + icon and choose new room option
6. enter room name , choose workspace and set Visibility public
7. click on create room button
8. go to #room
8. click on header to  see details
9. click on Members
10. check the members in list

## Expected Result:
all members should not show who is member of Workspaces ( after switching to online it's working fine)

## Actual Result:
 all members which is invited to workspace is added

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** Dev: 1.3.37-2
**Reproducible in staging?:** no
**Reproducible in production?:** no
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/fead72a2-a96a-4c34-8bf7-89cda709f607

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688628721649699

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011f674f9d8627617b</li>
        <li>Upwork Job ID: 1681233013381730304</li>
        <li>Last Price Increase: 2023-07-18</li>
    </ul>
</details>

**Comment by francoisl** [comment]
Slightly confused, I can reproduce the issue in production and dev, but not on staging.

<img width="1499" alt="image" src="https://github.com/Expensify/App/assets/2229301/7a558445-2c43-43e5-b050-5887076e5987">

At first glance, it looks like the code around this hasn't changed on something that was merged recently, but doesn't exist on staging. Going to review and test this [proposal](https://github.com/Expensify/App/issues/22353#issuecomment-1624128740) now.

**Comment by francoisl** [comment]
@kushu7 your proposal makes sense, but do you have an idea where the regression comes from? The code you're suggesting we change is from https://github.com/Expensify/App/pull/21072, which is already on production. 

I would have expected something like:

1. There was initially a bug where we exposed the full workspace members list for public rooms
2. We fixed it in https://github.com/Expensify/App/pull/21072
3. Another PR caused the current regression. At the moment, that PR would be merged on `main`, but not deployed to staging yet (because the issue can't be reproduced on staging)

**Comment by francoisl** [comment]
Could also be a backend change but I'm not seeing any recent change that could explain the issue.

Anyway, let's go with your proposal @kushu7 🚀

**Comment by francoisl** [comment]
@kushu7 FYI I went ahead and made the changes for your proposal since you were offline and this would be a deploy blocker, so you don't need to submit a PR. The PR is [here](https://github.com/Expensify/App/pull/22385), we'll still compensate you for the proposal.

**Comment by francoisl** [comment]
Going to remove the DeployBlocker label so that it doesn't block the deploy. The fix doesn't need to be CPed to staging since the error was only on dev.

**Comment by Julesssss** [comment]
Thanks @francoisl

**Comment by greg-schroeder** [comment]
Awaiting deploy to prod

**Comment by greg-schroeder** [comment]
Working on payment steps

**Comment by greg-schroeder** [comment]
Offers sent @gadhiyamanan @kushu7 

@Santhosh-Sellavel you'll take care of payment via NewDot request - Just awaiting you to complete the checklist

**Comment by gadhiyamanan** [comment]
@greg-schroeder accepted, thanks!

**Comment by greg-schroeder** [comment]
Okay payments made. Just waiting for the checklist from @Santhosh-Sellavel :)

**Comment by Santhosh-Sellavel** [comment]
@francoisl I Had already hovered over part checklist [here](https://github.com/Expensify/App/issues/22353#issuecomment-1624325453) about the PR which caused or missed this case.

> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [ ]  [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Do we need a discussion for this @francoisl?
> * [x]  [@Santhosh-Sellavel] Determine if we should create a regression test for this bug. Yes
> * [ ]  [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
> * [ ]  [@greg-schroeder] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
Bug: 

Room - Settings - Non-members appears in the list of room in offline mode

### Regression Steps:

1. Create a Workspace, and invite at least one additional person to it
2. Go offline
3. Click on the + floating button > New Room
4. Enter a room name, choose the new workspace, set the visibility to Public and create the room
5. Click on the room's header to see the details
6. Click on the Members list
7. Make sure you only see yourself in the Members list
8. Repeat the operation by creating a Private room, make sure you also only see yourself in the members list
9. Repeat the operation by creating a Restricted room, and make sure you can also see the other workspace members in the room's members list

👍  or 👎 

cc: @francoisl @greg-schroeder

**Comment by Santhosh-Sellavel** [comment]
@greg-schroeder 

The checklist is done, waiting for a review from @francoisl!

About the payment, does this qualify for the bonus as we hired an external contributor? 

Let me know the amount in a comment, so I can proceed with a request on ND, thanks!

**Comment by francoisl** [comment]
> The checklist is done, waiting for a review from @francoisl!

Reviewed, looks good to me.

> About the payment, does this qualify for the bonus as we hired an external contributor?

Did we hire an external contributor? We used @kushu7's proposal, who's not an external contributor as far as I know. Let me know if I'm missing something.

**Comment by greg-schroeder** [comment]
I didn't pay the bonus because we just used the proposal and an Expensify engineer actually implemented the code. To me that seemed most logical as the bonus is there to incentivize speed of implementation, which @kushu7 wasn't involved in beyond the initial idea stage.

Does that make sense?

**Comment by francoisl** [comment]
> >We used @kushu7's proposal, who's not an external contributor as far as I know
>
> @francoisl I'm external contributor 😅


Huh sorry guys, it's Friday and I can't think straight 😁😵‍💫 For some reason I was thinking we were talking about _agency_ contributors 😑

Anyway, @greg-schroeder's explanation makes sense to me regarding the bonus.

**Comment by greg-schroeder** [comment]
Okay filing for regression test. Closing!

**Comment by Santhosh-Sellavel** [comment]
I see a bonus not applicable based on the [comment](https://github.com/Expensify/App/issues/22353#issuecomment-1646320426).

So Requested a $1K Job fee alone. @greg-schroeder cc: @francoisl

**Comment by JmillsExpensify** [comment]
@greg-schroeder Can you please summarize the appropriate individual payments for all parties involved in this issue, per this new compliance process [in Slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1690321157273869)?

**Comment by greg-schroeder** [comment]
Hi @JmillsExpensify 

C - @kushu7 - $1,000
C+ - @Santhosh-Sellavel  - $1,000
Reporter - @gadhiyamanan - $250

**Comment by JmillsExpensify** [comment]
Reviewed details for @Santhosh-Sellavel. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.
